src/consumer.rs

```diff
@@ -1550,4 +1630,95 @@ mod tests {
 
         rt.block_on(f);
     }
+
+    #[test]
```

Comment:
In https://github.com/wyyerd/pulsar-rs/pull/112 I went through and cleaned up these tests to use `#[tokio::test]` - I think it makes them easier to read / maintain. Might be worth taking a look and doing that here as well.